claims_hosp/delphi_claims_hosp/modify_claims_drops.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+"""Modify the drops.
+
+Drops are expected to be numbered as:
+
+../EDI_AGG_INPATIENT/EDI_AGG_INPATIENT_1_07052020_1456.csv.gz
+../EDI_AGG_INPATIENT/EDI_AGG_INPATIENT_2_07052020_1456.csv.gz
+... etc.
+"""
+
+# standard
+from pathlib import Path
+
+# third party
+import numpy as np
+import pandas as pd
+
+
+def modify_and_write(data_path, logger, force=True):
+    """
+    Modify drops given a folder path.
+
+    Will rename necessary columns in the input files, and check the number of
+    columns and duplications.
+
+    Args:
+      data_path: path to the folder with duplicated drops.
+      force: if aggregated file exists, whether to overwrite or not
+
+    """
+    files = np.array(list(Path(data_path).glob("*.csv.gz")))
+
+    for f in files:
+        filename = str(f)
+        out_path = f.parents[0] / f.name
+        dfs = pd.read_csv(f, dtype={"PatCountyFIPS": str,
+                                    "patCountyFIPS": str})
+        if "servicedate" in dfs.columns:
+            dfs.rename(columns={"servicedate": "ServiceDate"}, inplace=True)
+        if "patCountyFIPS" in dfs.columns:
+            dfs.rename(columns={"patCountyFIPS": "PatCountyFIPS"}, inplace=True)
+        if "patHRRname" in dfs.columns:
+            dfs.rename(columns={"patHRRname": "Pat HRR Name"}, inplace=True)
+        if "patAgeGroup" in dfs.columns:
+            dfs.rename(columns={"patAgeGroup": "PatAgeGroup"}, inplace=True)
+        if "patHRRid" in dfs.columns:
+            dfs.rename(columns={"patHRRid": "Pat HRR ID"}, inplace=True)
+
+        assert np.sum(
+            dfs.duplicated(subset=["ServiceDate", "PatCountyFIPS",
+                                   "Pat HRR Name", "PatAgeGroup"])) == 0, \
+            f'Duplication across drops in {filename}!'
+        assert dfs.shape[1] == 10, f'Wrong number of columns in {filename}'
+
+        if force:
+            dfs.to_csv(out_path, index=False)
+            logger.info(f"Wrote {out_path}")
```

Comment:
Why is it needed for unit tests?

Reply:
Because I want the unit test to check the renaming results but not overwrite the raw files for the `./tests./test_data`

Reply:
ah okay. in that case, any of the following would be fine:

1. change the parameter name and documentation to describe what it actually does. "force" implies overriding a failed check, which is not what is happening here.
2. mock out `to_csv` to verify behavior in unittests instead of writing/overwriting on the filesystem. @dshemetov would you paste in your example?

Reply:
```py
# testA.py
import pandas as pd

def foo():
    y = pd.DataFrame.from_dict({'a': [5], 'b': [3]})
    return(y.to_csv())

# testB.py
import mock
from pandas import DataFrame
from testA import foo

def fake_to_csv(self):
    return "fake"

with mock.patch.object(DataFrame, 'to_csv', fake_to_csv):
    print(foo())

# python testB.py output
fake
```

From [here](https://stackoverflow.com/questions/5036920/mocking-out-methods-on-any-instance-of-a-python-class/5044894#5044894).